DATASETS_v2.md

```diff
@@ -0,0 +1,244 @@
+- `scale`
+  * How many of the digits are to the right of the decimal point.
+For example, the number "1234.5678" can be stored in a numeric type with a precision of 8 and a scale of 4.
+
+###### Extra type info for `"dataType": "timestamp"`
+- `timeZone`
+  * Eg: `"timeZone": "Europe/London"`
+  * The timezone that the timestamp is measured in.
+  * Can also be a UTC offset eg "+12:00", or "Z" for "+00:00"
+  * Can also be null for an unspecified timezone.
+
+#### `meta/legend/...`
+
+The legend folder of the dataset contains data known as "legends" that are used when reading features. Features that are written using one schema could be read later once the schema has changed. A legend contains the minimal amount of information required to adapt the feature to the current schema. This information is just the list of column IDs from the schema at the time of writing. For example, if features were written using the schema in the section above, this would also result in the following legend being written:
+
+```json
+[
+  "500f5ecc-c02c-e8db-052a-84efb4d08a00",
+  "b8ae8ff3-1691-7f6f-bbe0-c03c648b6d67",
+  "4324e825-ffed-8aff-ceac-53a7dde7e2f7",
+  "878f8e4e-433a-b7bb-74d5-b360ccfb3607",
+  "c8e75111-0506-a898-4d0e-ed1aa8c81280",
+]
+```
+
+Another legend entry is added whenever an update to the schema is committed, and they are never modified or deleted. They are part of the internal structure of the dataset and they need not be viewed by the end user.
+
+Each legend has a unique filename based on its hash. Legends are not stored using JSON, but an binary encoding called [msgpack](https://msgpack.org/index.html) that has equivalent capabilities.
+
+#### `meta/crs/{identifier}.wkt`
+A dataset may contain well-known-text CRS definitions of any of the CRS that are needed to interpret its geometry. The identifier that is part of the filename here should be the same as the `geometryCRS` identifier in the schema.
+
+#### `.sno-table/meta/metadata/dataset.json`
+A dataset may contain other structured metadata files that apply to the entire dataset, for instance XML metadata according to the http://www.isotc211.org/2005/gmd schema. These are stored in the `metadata/dataset.json` file, where they are keyed first by the metadata standard that is being adhered to, and then by the mimetype of the file. 
+
+Here is a truncated example of a `dataset.json` file with only one entry:
+```json
+{
+  "http://www.isotc211.org/2005/gmd": {
+    "text/xml": "<gmd:MD_Metadata …"
+  }
+}
+```
+
+### Features
+
+Every database table row is stored in its own file. It is stored as an array of values plus the name of the legend that should be used to read it. This array is serialised using messagepack (a binary format equivalent to JSON), but for the sake of readability, the example below is shown in JSON.
+
+For instance, a single feature might be stored as the following:
+```json
+[
+  "204b9886d5dbd9fe3a7edb9a7a7dba699b5202f7",  # legend
```

Comment:
can't have comments in json, so most formatters will object to this. IMO you don't need to have a comment in the json blob itself

Reply:
Done